docs/configuration-ini.md

```diff
@@ -0,0 +1,335 @@
+	
+	```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
+
+	
+4. ```testdata = <string>```
+
+    test data file path.   The data must be saved as following format
+	
+	```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
+	
+
+5. ```labeldata = <string>```
+
+    label data file path.   The data must be saved as following format
+	
+	
+
+### Layer Section
+Describe hyper-parameters for layer. Other is matter and it start from top to bottom.
+
+Start with "[ ${layer name} ]"
+
+1. ```type = <string>```
+
+   Type of Layer
+     - input : input layer
+     - conv2d : 2D convolution layer
+     - pooling2d : 2D pooling layer
+     - flatten : flatten layer
+     - fully_connected : fully connected layer
+     - batch_normalization : batch normalization layer
+     - activation : activation layer
+	 
+  
+2. ```kernel_size = <unsigned int>,<unsigned int>```
+
+   Kernel size for convolution layer
+   
+  
+3. ```bias_init_zero = <bool>```
+
+   token to set bias zero
+   
+  
+4. ```normalization = <bool>```
+
+   token to enable/disable normalization
+   
+  
+5. ```standardization = <bool>```
+
+   token to enable/disable standardization <bool>
+   
+  
+6. ```input_shape = <unsigned int>:<unsigned int>:<unsigned int>```
+
+   shape of input
+   it shouldn't be zero.
+   
+  
+7. ```activation = <string>```
+
+   set activation layer <string>
+     - tanh : tanh function
+     - sigmoid : sigmoid function
+     - relu : relu function
+     - softmax : softmax function
+	 
+  
+8. ```weight_decay = <string>```
+
+   set weight decay 
+     - l2norm : L2Norm
+	 
+  
+9. ```weight_decay_lambda = <float>```
+
+   coefficient for weight decay
+   
+  
+10. ```unit = <unsigned int>```
+
+     set the output layer for fully connected layer 
+   
+  
+11. ```weight_ini = <string>```
+
+     set weight initialization method 
+       - lecun_normal : LeCun normal initialization
+       - lecun_uniform : LeCun uniform initialization
+       - xavier_normal : xavier normal iniitalization
+       - xavier_uniform : xavier uniform initialization
+       - he_normal : He normal initialization
+       - he_uniform : He uniform initialization
+	 
+  
+12. ```filter = <unsigned int>```
+
+     set filter size for convolution layer
+   
+  
+13. ```stride = <unsigned int>,<unsigned int>```
+
+     set stride for convolution and pooling layer
+   
+  
+14. ```padding = <unsigned int>,<unsigned int>``` 
+
+     set padding for convolution and pooling layer
+   
+  
+15. ```pooling_size = <unsigned int>,<unsigned int>```
+
+     set pooling size for pooling layer
+   
+  
+16. ```pooling = <string>```
+
+     define type of pooling
+       - max : max pooling
+       - average : average pooling
+       - global_max : global max pooling
+       - global_average : global average pooling
+	 
+  
+17. ```flatten = <bool>```
+
+     another option to define flatten layer.
```

Comment:
Lets not details on how this is implemented.
IMO the description should be - "flattens the output of this layer. Enabling this option is equivalent to attaching a flatten layer after the current layer."